### PR TITLE
pdf2txt: clean up construction of LAParams from arguments

### PR DESCRIPTION
**Pull request by 0xabu**
This PR cleans up the way pdf2txt constructs the `LAParams` object it passes to `extract_text_to_fp`.

Specifically:
 1. Rather than hardcoding the default values of the parameters in two places, have pdf2txt fetch them from the object's existing defaults (all were identical, presumably intentionally so).
 2. Rather than passing the parameters as arguments to an intermediate function, construct LAParams closer to the argument parser.
 3. Call the LAParams constructor rather than setting attributes after construction, so that `LAParams._validate` runs to validate the chosen parameters.

As a deliberate side-effect of point 2 above, this fixes the following crash:
```
$ pdf2txt.py --boxes-flow=disabled test.pdf
Traceback (most recent call last):
  File "tools/pdf2txt.py", line 204, in <module>
    sys.exit(main())
  File "tools/pdf2txt.py", line 198, in main
    outfp = extract_text(**vars(A))
  File "tools/pdf2txt.py", line 66, in extract_text
    pdfminer.high_level.extract_text_to_fp(fp, **locals())
  File "pdfminer/high_level.py", line 85, in extract_text_to_fp
    interpreter.process_page(page)
  File "pdfminer/pdfinterp.py", line 896, in process_page
    self.device.end_page(page)
  File "pdfminer/converter.py", line 51, in end_page
    self.cur_item.analyze(self.laparams)
  File "pdfminer/layout.py", line 822, in analyze
    group.analyze(laparams)
  File "pdfminer/layout.py", line 575, in analyze
    LTTextGroup.analyze(self, laparams)
  File "pdfminer/layout.py", line 362, in analyze
    obj.analyze(laparams)
  File "pdfminer/layout.py", line 575, in analyze
    LTTextGroup.analyze(self, laparams)
  File "pdfminer/layout.py", line 362, in analyze
    obj.analyze(laparams)
  File "pdfminer/layout.py", line 575, in analyze
    LTTextGroup.analyze(self, laparams)
  File "pdfminer/layout.py", line 362, in analyze
    obj.analyze(laparams)
  File "pdfminer/layout.py", line 577, in analyze
    self._objs.sort(
  File "pdfminer/layout.py", line 578, in <lambda>
    key=lambda obj: (1 - laparams.boxes_flow) * obj.x0
TypeError: unsupported operand type(s) for -: 'int' and 'str'
```
Thus, this supersedes PR #657 (which was a minimal fix for just that crash).

Related: Issue #477, PR #479

Fix https://github.com/pdfminer/pdfminer.six/issues/540

**How Has This Been Tested?**

 * I ran `pdf2txt --boxes_flow=disabled`, and verified that (a) there was no crash, and (b) the alternate layout algorithm was used in `LTLayoutContainer.analyze`.
 * I verified that changing layout parameters alters the output.
 * I verified that passing invalid layout parameters now raises an error:
 ```
$ tools/pdf2txt.py --boxes-flow 42 samples/simple1.pdf
Traceback (most recent call last):
  File "tools/pdf2txt.py", line 217, in <module>
    sys.exit(main())
  File "tools/pdf2txt.py", line 199, in main
    A = parse_args(args)
  File "tools/pdf2txt.py", line 189, in parse_args
    parsed_args.laparams = LAParams(**kwargs)
  File "pdfminer/layout.py", line 89, in __init__
    self._validate()
  File "pdfminer/layout.py", line 100, in _validate
    raise ValueError(boxes_flow_err_msg)
ValueError: LAParam boxes_flow should be None, or a number between -1 and +1
```

**Checklist**

- [x] I have added tests that prove my fix is effective or that my feature 
  works
- [x] I have added docstrings to newly created methods and classes
- [x] I have optimized the code at least one time after creating the initial 
  version
- [x] ~~I have updated the [README.md](../README.md) or~~ I have verified that this
  is not necessary
- [x] ~~I have updated the [readthedocs](../docs/source) documentation or~~ I 
  verified that this is not necessary
- [x] I have added a concise human-readable description of the change to 
  [CHANGELOG.md](../CHANGELOG.md)

### CODE REVIEW

File: tools/pdf2txt.py [hunk 1]
```diff
@@ -32,13 +30,7 @@ def float_or_disabled(x: str) -> FloatOrDisabled:
...
     files: Iterable[str] = [],
     outfile: str = '-',
-    no_laparams: bool = False,
-    all_texts: Optional[bool] = None,
-    detect_vertical: Optional[bool] = None,
-    word_margin: Optional[float] = None,
-    char_margin: Optional[float] = None,
-    line_margin: Optional[float] = None,
-    boxes_flow: Optional[FloatOrDisabled] = None,
+    laparams: Optional[LAParams] = None,
```
Comment (pietermarsman) [inline tools/pdf2txt.py:33]: Breaking change.

This method shouldn't be used directly but many people will probably do so. 

Did a quick search and seems like this is not the case: https://github.com/search?p=1&q=pdfminer.tools.pdf2txt&type=Code

So, we're fine here! :)

File: tools/pdf2txt.py [hunk 2]
```diff
@@ -194,16 +175,28 @@ def maketheparser() -> argparse.ArgumentParser:
...
+
+    # Propagate parsed layout parameters to LAParams object
+    if parsed_args.no_laparams:
+        parsed_args.laparams = None
+    else:
+        kwargs = {}
+        for param in ("all_texts", "detect_vertical", "word_margin",
+                      "char_margin", "line_margin", "boxes_flow"):
+            kwargs[param] = getattr(parsed_args, param)
+        parsed_args.laparams = LAParams(**kwargs)
```
Comment (pietermarsman) [inline tools/pdf2txt.py:189]: I would prefer a more direct approach for readability.

**Review by pietermarsman — APPROVED** [review]
Thanks so much! Again a very nice improvement! 

Will do some small changes myself and then merge it.

**Comment by pietermarsman** [comment]
I did some more changes:
- Not using `locals()` or `getattr()` anymore when creating `LAParams`.
- Moved all post-processing of the parsed args to the `parse_args()` method. 
- Also used the default values from `LAParams` for `--detect-vertical` and `--all-texts`. Presumably you omited those by accident? 
- Add cli argument for `--line-overlap` as well.

**Comment by 0xabu** [comment]
LGTM, thanks.

**Pull request merged**